[2020-12-22 14:21:51,302] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:21:51,324] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:21:51,327] {taskinstance.py:866} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:21:51,330] {taskinstance.py:867} INFO - Starting attempt 48 of 48
[2020-12-22 14:21:51,330] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:21:51,350] {taskinstance.py:887} INFO - Executing <Task(LivyBatchOperator): preprocess_data> on 2020-12-22T12:43:23.210460+00:00
[2020-12-22 14:21:51,355] {standard_task_runner.py:53} INFO - Started process 61860 to run task
[2020-12-22 14:21:51,438] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [running]> dcb43f18a24b
[2020-12-22 14:21:51,464] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,464] {batch.py:214} INFO - Submitting the batch to Livy... Payload:
{
  "file": "s3a://alanchn31-loan-default-prediction/dist/main.py",
  "args": [
    "--job",
    "preprocess_data",
    "--phase",
    "train",
    "--mode",
    "aws",
    "--awsKey",
    "AKIAUW4FKHTMR2UR3INH",
    "--awsSecretKey",
    "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",
    "--s3Bucket",
    "alanchn31-loan-default-prediction"
  ],
  "pyFiles": [
    "s3a://alanchn31-loan-default-prediction/dist/src.zip"
  ],
  "name": "preprocess_data_scheduled__2020-12-22T12:43:23.210460+00:00"
}
[2020-12-22 14:21:51,474] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,473] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:21:51,477] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,477] {http_hook.py:136} INFO - Sending 'POST' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches
[2020-12-22 14:21:51,497] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,497] {batch.py:166} INFO - Batch successfully submitted with id = 44.
[2020-12-22 14:21:51,500] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,500] {batch.py:80} INFO - Getting batch 44 status...
[2020-12-22 14:21:51,511] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,511] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:21:51,514] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,514] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/44
[2020-12-22 14:21:51,521] {logging_mixin.py:112} INFO - [2020-12-22 14:21:51,521] {batch.py:90} INFO - Batch 44 has not finished yet (state is 'starting')
[2020-12-22 14:22:11,542] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,542] {batch.py:80} INFO - Getting batch 44 status...
[2020-12-22 14:22:11,549] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,549] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:22:11,553] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,553] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/44
[2020-12-22 14:22:11,561] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,561] {batch.py:299} INFO - --------------------------------------------------Full log for batch 44--------------------------------------------------
[2020-12-22 14:22:11,570] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,570] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:22:11,574] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,573] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/44/log?from=0&size=100
[2020-12-22 14:22:11,581] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,581] {batch.py:309} INFO - stdout: 
[2020-12-22 14:22:11,582] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,581] {batch.py:309} INFO - 2020-12-22 14:21:53,174 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2020-12-22 14:22:11,582] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,582] {batch.py:309} INFO - Traceback (most recent call last):
[2020-12-22 14:22:11,582] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,582] {batch.py:309} INFO -   File "/tmp/spark-d49e1142-f811-4128-aacf-0c0c1566b2b7/main.py", line 87, in <module>
[2020-12-22 14:22:11,583] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,582] {batch.py:309} INFO -     main()
[2020-12-22 14:22:11,583] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,583] {batch.py:309} INFO -   File "/tmp/spark-d49e1142-f811-4128-aacf-0c0c1566b2b7/main.py", line 68, in main
[2020-12-22 14:22:11,583] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,583] {batch.py:309} INFO -     spark_logger.warn('no config file found')
[2020-12-22 14:22:11,583] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,583] {batch.py:309} INFO - NameError: name 'spark_logger' is not defined
[2020-12-22 14:22:11,584] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,584] {batch.py:309} INFO - 2020-12-22 14:21:56,062 INFO util.ShutdownHookManager: Shutdown hook called
[2020-12-22 14:22:11,584] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,584] {batch.py:309} INFO - 2020-12-22 14:21:56,063 INFO util.ShutdownHookManager: Deleting directory /tmp/spark-d49e1142-f811-4128-aacf-0c0c1566b2b7
[2020-12-22 14:22:11,584] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,584] {batch.py:309} INFO - 2020-12-22 14:21:56,068 INFO impl.MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2020-12-22 14:22:11,584] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,584] {batch.py:309} INFO - 2020-12-22 14:21:56,069 INFO impl.MetricsSystemImpl: s3a-file-system metrics system stopped.
[2020-12-22 14:22:11,584] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,584] {batch.py:309} INFO - 2020-12-22 14:21:56,069 INFO impl.MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2020-12-22 14:22:11,585] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,585] {batch.py:309} INFO - 
stderr: 
[2020-12-22 14:22:11,585] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,585] {batch.py:309} INFO - 
YARN Diagnostics: 
[2020-12-22 14:22:11,586] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,586] {batch.py:309} INFO - spark-submit start failed
[2020-12-22 14:22:11,586] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,586] {batch.py:318} INFO - --------------------------------------------------End of full log for batch 44--------------------------------------------------
[2020-12-22 14:22:11,586] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,586] {batch.py:335} INFO - Closing batch with id = 44
[2020-12-22 14:22:11,595] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,594] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:22:11,595] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,595] {http_hook.py:136} INFO - Sending 'DELETE' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/44
[2020-12-22 14:22:11,603] {logging_mixin.py:112} INFO - [2020-12-22 14:22:11,603] {batch.py:340} INFO - Batch 44 has been closed
[2020-12-22 14:22:11,612] {taskinstance.py:1128} ERROR - Batch 44 failed with state 'dead'
Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 966, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 173, in execute
    ).execute(context)
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/sensors/base_sensor_operator.py", line 107, in execute
    while not self.poke(context):
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 96, in poke
    raise AirflowException(f"Batch {self.batch_id} failed with state '{state}'")
airflow.exceptions.AirflowException: Batch 44 failed with state 'dead'
[2020-12-22 14:22:11,615] {taskinstance.py:1170} INFO - All retries failed; marking task as FAILED.dag_id=loan_prediction_train_model, task_id=preprocess_data, execution_date=20201222T124323, start_date=20201222T142151, end_date=20201222T142211
[2020-12-22 14:22:16,350] {logging_mixin.py:112} INFO - [2020-12-22 14:22:16,350] {local_task_job.py:103} INFO - Task exited with return code 1
